packages/flutter/lib/src/widgets/async.dart

```diff
@@ -520,20 +611,44 @@ class FutureBuilder<T> extends StatefulWidget {
   /// Creates a widget that builds itself based on the latest snapshot of
   /// interaction with a [Future].
   ///
-  /// The [builder] must not be null.
+  /// The [initialData] argument specifies the data that will be used to create
+  /// the snapshots provided to [builder] until a non-null [future] has
+  /// completed.
+  ///
+  /// The [builder] argument must not be null.
   const FutureBuilder({
     Key key,
     this.future,
-    this.initialData,
+    @required T initialData,
+    @required this.builder,
+  }) : assert(builder != null),
+       hasInitialData = true,
+       _initialData = initialData,
+       super(key: key);
+
+  /// Creates a widget that builds itself based on the latest snapshot of
+  /// interaction with a [Future].
+  ///
+  /// The snapshots that will be provided to [builder] until a non-null
+  /// [future] has completed will contain no data.
+  ///
+  /// The [builder] argument must not be null.
+  const FutureBuilder.withoutInitialData({
```

Comment:
isn't this the common case? or even the only case, usually? seems weird to make this harder to use.